pkg/streamingpromql/operators/deduplicate_and_merge.go

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package operators
+
+import (
+	"context"
+	"errors"
+	"slices"
+
+	"github.com/prometheus/prometheus/promql/parser/posrange"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+var errVectorContainsMetricsWithSameLabels = errors.New("vector cannot contain metrics with the same labelset")
+
+type DeduplicateAndMerge struct {
```

Comment:
(nit) should this be `DeduplicateAndMergeOperator` since it implements the operator interface?

Reply:
None of the other operators have an `Operator` suffix - eg. `Aggregation`, `FunctionOverInstantVector` etc.

Adding an `Operator` suffix would introduce a stutter when the type is written with the package name, eg. `operators.DeduplicateAndMerge`.

Reply:
I confused myself with BinaryOperation :-p